### ISSUE TITLE
Search does not work

### ISSUE THREAD

**Issue by trinix1975** [olejon/spotcommander#90]
Hi,

When I try to use search with any words from my iPhone or from another computer I get the following message after a few seconds.

"Could not get search results. Try again."

My server is Linux Mint 17.1 with kernel 3.13.0-37-generic

Thanks

**Comment by olejon** [comment]
Is Search the only online function that doesn't work? Works fine here. Does Browse work? Does browsing playlists, albums and artist work?

**Comment by trinix1975** [comment]
Hi,

Thanks for the answer. Your program is awesome by the way, very nice interface. I can browse from the "Browse" menu with a cloud next to it. I can see my playlist I made from the Spotify version.

However I'm not sure what you mean by albums and artist. I don't see such categories along with Playlists, Library, Browse and Search. But if you mean if I search for example for "Sting" in the Search section, that doesn't work.

**Comment by olejon** [comment]
Thanks! I mean if you for instance can go into a playlist, tap on a song and then tap on the "person" icon (artist), and see if it works. Or the song overflow menu and then Go to Album.

**Comment by trinix1975** [comment]
I see what you mean. Yes I can access the artists info and all that. Thanks.

**Comment by trinix1975** [comment]
Ha, check this out. I clicked on "Clear Cache" and "Remove all playlists" from the settings menu in Spotcommander and it's now working. Good to know :)

**Comment by olejon** [comment]
OK, I was going to ask you to clear the cache. It caches searches for a while so they don't take so long the next time, but if something happened it may have cached the incorrect data.

**Comment by trinix1975** [comment]
Can we deactivate the cache. It still happens randomly and I cannot find when and what I did to break it. It might when the computer goes to sleep or something like that. I'll let you know when I can reproduce the problem at will.

**Comment by olejon** [comment]
Ask you can see here in code it shouldn't cache the results if they don't return the correct metadata:

https://github.com/olejon/spotcommander/blob/master/spotcommander/functions.php#L1315

You can comment out that line (1315 in functions.php) and clear the cache to disable cache completely for searches.

**Comment by olejon** [comment]
Weird. Take a look in `/var/log/apache2/error.log`

**Comment by trinix1975** [comment]
Nothing that seems related. This didn't happen during the time I did a search. Most of the time, I search for something, the arrow is spinning for around 10 sec. Then it says "Could not get search results. Try again.". Then I have to press either once or twice the refresh circular arrow to get the result.

[Tue Sep 01 09:16:24.554275 2015] [mpm_prefork:notice] [pid 3249] AH00163: Apache/2.4.7 (Ubuntu) PHP/5.5.9-1ubuntu4.11 configured -- resuming normal operations
[Tue Sep 01 09:16:24.554304 2015] [core:notice] [pid 3249] AH00094: Command line: '/usr/sbin/apache2'

**Comment by olejon** [comment]
It's hard to explain as search uses the same API as all the other available Spotify web APIs. Does it work in the API console?

https://developer.spotify.com/web-api/console/get-search-item/

OAuth Token is not necessary for search.

**Comment by trinix1975** [comment]
It's working fine on the Spotify site. I took the time to analyse a bit how it's coded and I deactivated the local database requests. I thought it worked better but I still had the same behavior. Then I limited the number of queries to Spotify to only get the tracks, artists and album (no more playlists and users). Now it's a bit faster but I still have to press the refresh a couple of times when it's stuck.

When I have time I'll try to isolate the problem. Actually I should probably try to do only one request to Spotify to see if it's doing more than one at the same time that goes wrong... although why am I the only who has this issue, that's another story.
